File: frontend/src/backend/shoppingService.js
import {config, appFetch} from './appFetch';

export const buy = (sessionId, numberOfTickets, creditCard, onSuccess, onErrors) => {

    let path = `/buying/buy/${sessionId}`;

    appFetch(path, config('POST', {numberOfTickets, creditCard}), onSuccess, onErrors);
}

export const findOrders = ({page}, onSuccess) =>
    appFetch(`/buying/orders?page=${page}`, config('GET'), onSuccess);

export const deliverTickets = (creditCard, orderId, onSuccess, onErrors) =>
    appFetch(`/buying/deliver`, config('POST', {creditCard,orderId}), onSuccess, onErrors);

File: backend/src/main/java/es/udc/paproject/backend/model/services/ShoppingService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Order;
import es.udc.paproject.backend.model.exceptions.*;

public interface ShoppingService {

    Order buy(Long userId, Long sessionId, String CreditCard, int numberOfTickets) throws InstanceNotFoundException, IncorrectQuantityOfTickets, SessionAlreadyStartedException;
    void deliverTickets(Long orderId, String creditCard) throws InstanceNotFoundException, SessionAlreadyStartedException, AlreadyDeliveredException, NotMatchingCreditCard;
    Block<Order> findOrders(Long userId, int page, int size);

}
